api/src/game_state_server.py

```diff
@@ -156,93 +104,118 @@ async def handle_connection(
         session_id = str(uuid4())
         with timber.context(connection={'session_id': session_id, 'room_id': room_id}):
             async with self._rate_limiter.rate_limited_connection(client_ip, room_id):
-                room_changes = await self.room_store.changes(room_id)
+                listener_q: asyncio.Queue[Response] = asyncio.Queue()
 
                 with self.apm_transaction('connect'):
-                    result = await self.room_store.apply_mutation(
-                        room_id,
-                        request_id=None,
-                        mutate=lambda entities: self._acquire_room_slot(
-                            room_id, entities, client_ip
-                        ),
+                    if not self._room_context_by_id.get(room_id):
+                        await self._acquire_room_slot(room_id, client_ip)
+                        room = create_room(room_id, await self.room_store.read(room_id))
```

Comment:
We need to ensure that we've successfully subscribed to changes before we read the room so we don't miss updates

Reply:
Good catch.